src/mat/vplast/4C_mat_vplast_reform_johnsoncook.cpp

```diff
@@ -137,13 +146,26 @@ Mat::Viscoplastic::ReformulatedJohnsonCook::evaluate_derivatives_of_plastic_stra
   // then we check the yield condition
   if (evaluate_stress_ratio(equiv_stress, used_equiv_plastic_strain) >= 1.0)
   {
-    // compute the exact derivatives
-    equiv_plastic_strain_rate_ders(0, 0) =
-        p * std::exp(e * (equiv_stress / yield_strength - 1.0)) * e / yield_strength;
+    // compute first the logarithms of our derivatives (try to avoid overflow!)
+    double log_deriv_sigma = std::log(p) + e * (equiv_stress / yield_strength - 1.0) + std::log(e) -
+                             std::log(yield_strength);
+    double log_deriv_eps = std::log(p) + e * (equiv_stress / yield_strength - 1.0) + std::log(e) +
+                           std::log(equiv_stress) - 2.0 * std::log(yield_strength) + std::log(B) +
+                           std::log(N) + (N - 1.0) * std::log(used_equiv_plastic_strain);
+
+    // check overflow error using these logarithms
+    if ((!log_substep && (std::log(dt) + log_deriv_sigma > 10.0) &&
+            (std::log(dt) + log_deriv_eps > 10.0)) &&
+        (log_substep && (std::log(dt) + log_deriv_sigma > std::exp(10.0)) &&
+            (std::log(dt) + log_deriv_eps > std::exp(10.0))))
+    {
+      err_status = 2;
+      return Core::LinAlg::Matrix<2, 1>{true};
+    }
```

Comment:
You are computing the same expensive functions, `log` and `exp`, and sometimes longer subexpressiones like `std::log(p) + e * (equiv_stress / yield_strength - 1.0) + std::log(e)` multiple times. This results in way more instructions than you have to do and you could extract the subexpressions.

Reply:
Small example where I rearranged the computations to make this specific part 2.5 times faster(I did not even include the overflow check): https://quick-bench.com/q/ELEz4rotYR56n1y1IThkiFlZxWU

You don't need to do that in this PR. I just wanted to point out that this code looks really expensive and since you do this on the quadrature point I can imagine these calculations contribute significantly to your total run time.

Reply:
I have included your suggestions now. Thanks a lot for the hint and for the offline discussion!